ts/components/BottomSheetComponent.tsx

```diff
@@ -0,0 +1,182 @@
+    left: 0,
+    top: 0,
+    position: "absolute",
+    alignItems: "center",
+    justifyContent: "center"
+  }
+});
+
+const BottomSheetComponent: React.FunctionComponent<Props> = (props: Props) => {
+  const bottomSheetRef = React.useRef<BottomSheet>(null);
+  const [opacity] = React.useState(new Animated.Value(0));
+  const [accessible, setAccessible] = React.useState(false);
+
+  const color = opacity.interpolate({
+    inputRange: [0, 1],
+    outputRange: ["rgba(0,0,0,0)", "rgba(0,0,0,0.5)"]
+  });
+
+  const modalBackdrop = {
+    backgroundColor: color
+  };
+
+  const openBackdropAnimation = () =>
+    Animated.timing(opacity, {
+      useNativeDriver: false,
+      toValue: 1,
+      duration: 200
+    }).start();
+
+  const closeBackdropAnimation = () =>
+    Animated.timing(opacity, {
+      useNativeDriver: false,
+      toValue: 0,
+      duration: 200
+    }).start();
+
+  React.useEffect(() => {
+    if (props.content) {
+      openBottomSheet().catch(_ => closeBottomSheet());
+    } else {
+      closeBottomSheet();
+    }
+  }, [props.content]);
+
+  useHardwareBackButton(() => {
+    closeBottomSheet();
+    return true;
+  });
+
+  const closeBottomSheet = () => {
+    setAccessible(false);
+    if (bottomSheetRef && bottomSheetRef.current) {
+      bottomSheetRef.current.snapTo(0);
+      closeBackdropAnimation();
+      props.handleClose();
+    }
+  };
+
+  const openBottomSheet = async () => {
+    const isScreenReaderActive = await isScreenReaderEnabled();
+    setAccessible(isScreenReaderActive);
+    if (bottomSheetRef && bottomSheetRef.current) {
+      bottomSheetRef.current.snapTo(1);
+      openBackdropAnimation();
+    }
+  };
+
+  const renderHeader = () => (
+    <View style={[styles.row, styles.container]}>
+      <Text style={styles.title} semibold={true}>
+        {props.bottomSheetTitle}
+      </Text>
+      <ButtonDefaultOpacity
+        style={styles.modalClose}
+        onPress={closeBottomSheet}
+        transparent={true}
+        accessible={true}
+        accessibilityRole={"button"}
+        accessibilityLabel={I18n.t("global.buttons.close")}
+      >
+        <IconFont
+          name="io-close"
+          color={customVariables.lightGray}
+          style={styles.icon}
+        />
+      </ButtonDefaultOpacity>
+    </View>
+  );
+
+  return accessible ? (
+    <Modal>
+      <Container>
+        <AppHeader noLeft={true}>{renderHeader()}</AppHeader>
+        <View style={styles.content}>{props.content}</View>
+      </Container>
+    </Modal>
+  ) : (
+    <>
+      <Animated.View
+        pointerEvents={"none"}
+        style={[styles.hover, modalBackdrop]}
+      />
+      <BottomSheet
+        ref={bottomSheetRef}
+        snapPoints={[0, props.maxSnapPoint]}
+        onOpenEnd={openBackdropAnimation}
+        onCloseEnd={closeBackdropAnimation}
+        renderHeader={renderHeader}
+        renderContent={() => (
+          <View style={styles.content}>{props.content}</View>
+        )}
```

Comment:
```suggestion
        renderContent={() => (
          <View style={styles.content}>
            <SafeAreaView style={IOStyles.flex}>{props.content}</SafeAreaView>
          </View>
        )}
```
My proposal to avoid possible overflows in the safe area:
<img src=https://user-images.githubusercontent.com/26501317/96768376-1e7b5a00-13de-11eb-929a-adc3a511ccdd.png width=300 />